File: C programming/product of digits.c
#include <stdio.h>

int main() {
    int num, remainder, product = 1;

    printf("Enter a number: ");
    scanf("%d", &num);

    if (num == 0) {
        product = 0;
    }

    while (num != 0) {
        remainder = num % 10;
        product *= remainder;
        num /= 10;
    }

    printf("Product of digits = %d\n", product);
    return 0;
}

File: Java programming/Day-2/product of digits.java
import java.util.Scanner;

public class ProductOfDigits {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter an integer: ");
        int num = scanner.nextInt();
        
        int product = 1;
        int temp = Math.abs(num);
        
        if (temp == 0) {
            product = 0;
        } else {
            while (temp > 0) {
                product *= (temp % 10);
                temp /= 10;
            }
        }
        
        System.out.println("The product of digits in " + num + " is: " + product);
        scanner.close();
    }
}
